crates/snapbox/src/data.rs

```diff
@@ -112,37 +112,25 @@ impl Data {
     ///
     /// Note: this will **not** do a binary-content check
     pub fn make_text(&mut self) -> Result<(), std::str::Utf8Error> {
-        *self = Self::text(std::mem::take(self).into_string()?);
+        *self = Self::text(std::mem::take(self).try_string()?);
         Ok(())
     }
 
-    /// Coerce to a string
-    ///
-    /// Note: this will **not** do a binary-content check
-    pub fn into_string(self) -> Result<String, std::str::Utf8Error> {
-        match self.inner {
+    /// Tries to convert the underlying data into a String
+    pub fn try_string(&self) -> Result<String, std::str::Utf8Error> {
```

Comment:
As we add more formats, a single error type isn't sufficient.  We should probably make this an `Option<String>`